Ginger/Ginger/UserControlsLib/UCDataMapping.xaml.cs

```diff
@@ -0,0 +1,553 @@
+            }
+        }
+        private void MappedValuePropertyChanged(string mappedValueProperty)
+        {
+            if (MappedType == eDataType.OutputVariable.ToString()
+                || MappedType == eDataType.GlobalVariable.ToString()
+                || MappedType == eDataType.ApplicationModelParameter.ToString())
+            {
+                OnPropertyChanged(nameof(MappedValueGUID));
+            }
+            else
+            {
+                OnPropertyChanged(nameof(MappedValue));
+            }
+            
+            MarkMappedValueValidation();
+        }
+
+        private void MarkMappedValueValidation()
+        {
+            bool isValid = true;
+
+            if ((MappedType != eDataType.None.ToString() && MappedValue == string.Empty)
+                || (MappedType == eDataType.Variable.ToString() && !GingerCore.General.CheckComboItemExist(xVariablesComboBox, MappedValue))
+                || ((MappedType == eDataType.OutputVariable.ToString() || MappedType == eDataType.GlobalVariable.ToString() || MappedType == eDataType.ApplicationModelParameter.ToString()) && !GingerCore.General.CheckComboItemExist(xOptionalValuesComboBox, MappedValue, "Guid"))
+                || (MappedType == eDataType.DataSource.ToString() && GingerCoreNET.GeneralLib.General.CheckDataSource(MappedValue, WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<DataSourceBase>()) != string.Empty))               
+            {
+                isValid = false;
+            }
+
+            if (isValid == false)
+            {
+                this.BorderThickness = new Thickness(1);
+                this.BorderBrush = Brushes.Red;
+            }
+            else
+            {
+                this.BorderThickness = new Thickness(0);
+                this.BorderBrush = null;
+            }
+        }
+
+        private void xMappedTypeComboBox_DropDownOpened(object sender, EventArgs e)
+        {
+            _prevMappedType = MappedType;
+        }
+
+        private void xMappedTypeComboBox_DropDownClosed(object sender, EventArgs e)
+        {
+            if (_prevMappedType != MappedType)
+            {
+                //reset value between type selection
+                MappedValue = string.Empty;
+            }
+
+            if (MappedType == eDataType.None.ToString())
+            {
+                MappedValue = string.Empty;
+            }
+        }
+
+        private static void OnEnableDataMappingPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs args)
+        {
+            var control = sender as UCDataMapping;
+            if (control != null)
+            {
+                control.EnableDataMappingPropertyChanged((bool)args.NewValue);
+            }
+        }
+        private void EnableDataMappingPropertyChanged(bool enabelMapping)
+        {
+            EnableDataMapping = enabelMapping;
+            if (EnableDataMapping == true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)